### PR TITLE
[WIP] Mongodb otel refactor

### PR DESCRIPTION
**Pull request by vkn**
Initially #40191 added mongo db commands to otel traces. This PR moves the implementation from mongodb-client extension to opentelemetry

### CODE REVIEW

**Comment by vkn** [comment]
@brunobat I've added docs and parent-child relationship tests, which shows that in reactive case the mongo span is not added to the parent, as the listener is executed in a separate thread from vert.x event loop thread containing parent span. Any hints how to propagate the parent context to the listener. I've tried a few things but it didn't work
Thanks!

**Comment by brunobat** [comment]
Let me think about this...

File: integration-tests/opentelemetry-mongodb-client-instrumentation/src/main/java/io/quarkus/it/opentelemetry/ReactiveBookResource.java [hunk 1]
```diff
@@ -38,6 +40,14 @@ public CompletionStage<List<Book>> getBooks() {
         return getCollection().find().collect().asList().subscribeAsCompletionStage();
     }
 
+    @GET
+    @Path("/invalid")
+    public CompletionStage<List<Book>> getBooksError() {
+        BsonDocument query = new BsonDocument();
+        query.put("$invalidop", new BsonDouble(0d));
+        return getCollection().find(query).collect().asList().subscribeAsCompletionStage();
```
Comment (brunobat) [inline integration-tests/opentelemetry-mongodb-client-instrumentation/src/main/java/io/quarkus/it/opentelemetry/ReactiveBookResource.java:48]: The context propagation will not happen when calling the MongoDB directly. 
Please check the injected reactive client, it should provide the required propagation: https://quarkus.io/guides/mongodb#reactive.

**Review by brunobat — CHANGES_REQUESTED** [review]
Thanks for your time on this. 
Let's keep your original implementation and only add the documentation and tests to this PR.

File: extensions/opentelemetry/runtime/pom.xml [hunk 2]
```diff
@@ -26,6 +26,11 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-vertx</artifactId>
         </dependency>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-mongodb-client</artifactId>
```
[the remark this answers is not in the document]
Comment (vkn) [reply extensions/opentelemetry/runtime/pom.xml:31]: @brunobat Ok. I've opened #40714 . Canceling this one

File: integration-tests/opentelemetry-mongodb-client-instrumentation/src/main/java/io/quarkus/it/opentelemetry/ReactiveBookResource.java (hunk 1 above)
Comment (vkn) [reply integration-tests/opentelemetry-mongodb-client-instrumentation/src/main/java/io/quarkus/it/opentelemetry/ReactiveBookResource.java:56]: I've fixed reactive context propagation by a registering `com.mongodb.reactivestreams.client.ReactiveContextProvider`. The same should be done also for blocking client when mongo queries are executed in a thread pool, but thats would be another PR. My first attempt to implement `com.mongodb.client.SynchronousContextProvider` did not work, unfortunately

**Pull request closed without merging**